### PR TITLE
deps: Print cask dependencies 

### PR DESCRIPTION
**Pull request by whoiswillma**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

* Allow references to commands when using `brew deps`, e.g. `brew deps
cellery`
* Fix crash when using `brew deps <formula> --installed
--include-requirements`, e.g. `brew deps vale --installed --include-requirements`
* Do not include runtime dependencies as direct dependencies when using
`--tree`

   Otherwise, every recursive dependency is included as a direct dependency of the formula, which then prints out the tree for that direct dependency, cluttering up the output.

I added a class called `Dependent` that adapts casks for use as formulae in the existing deps methods.

### CODE REVIEW

File: Library/Homebrew/cmd/deps.rb [hunk 1]
```diff
@@ -211,4 +229,61 @@ def recursive_deps_tree(f, prefix, recursive)
 
     @dep_stack.pop
   end
+
+  class Dependent
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/deps.rb:233]: How about calling this `CaskDependent` and putting it in its own class somewhere?

File: Library/Homebrew/dependencies.rb [hunk 2]
```diff
@@ -123,3 +123,61 @@ def reject_ignores(dependables, ignores, includes)
     end
   end
 end
+
+# An adapter for casks to provide dependency information in a formula-like interface
+class CaskDependent
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dependencies.rb:128]: I'd be tempted to say a `cask_dependent` file or `cask/dependent` would be even nicer. A few unit tests then would be nice, too!

File: Library/Homebrew/cmd/deps.rb [hunk 3]
```diff
@@ -64,51 +65,68 @@ def deps
     Formulary.enable_factory_cache!
 
     recursive = !args.send("1?")
-    installed = args.installed? || args.formulae.all?(&:opt_or_installed_prefix_keg)
```
Comment (whoiswillma) [inline Library/Homebrew/cmd/deps.rb:67]: Was the intention of `opt_or_installed_prefix_keg` here just to check if a version was installed, or is there some subtlety that I'm missing?

File: Library/Homebrew/dependencies.rb [hunk 4]
```diff
@@ -123,3 +123,61 @@ def reject_ignores(dependables, ignores, includes)
...
+  end
+
+  def requirements
+    @requirements ||= begin
+      requirements = []
+      dsl_reqs = @cask.depends_on
+
+      dsl_reqs.arch&.each do |arch|
+        requirements << ArchRequirement.new([:x86_64]) if arch[:bits] == 64
+        requirements << ArchRequirement.new([arch[:type]])
```
Comment (whoiswillma) [inline Library/Homebrew/dependencies.rb:160]: Not sure if this is correct on my half? Based on `ArchRequirement::satisfy`, seems like `:x86_64` means 64-bit, but with arm macs right around the corner is this still correct?

File: Library/Homebrew/cmd/deps.rb (hunk 3 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/cmd/deps.rb:67]: Yeh, I think so. Could instead use `.any_version_installed?` I think here.

File: Library/Homebrew/dependencies.rb (hunk 4 above)
Comment (reitermarkus) [reply Library/Homebrew/dependencies.rb]: I don't think we have a single cask using `depends_on arch:` so not sure what this should look like.

Comment (MikeMcQuaid) [reply Library/Homebrew/dependencies.rb]:
```
Casks/locklizard-safeguard-viewer.rb
12:  depends_on arch: ":x86_64"
```

That's the only one. No `bits` references. I think you can punt on adding this architecture requirement for now, maybe?

Comment (reitermarkus) [reply Library/Homebrew/dependencies.rb]: > That's the only one.

And that one is even using it wrong with a string instead of symbol.

**Comment by MikeMcQuaid** [comment]
Great work again @whoiswillma!

**Comment by reitermarkus** [comment]
```
Error: uninitialized constant Homebrew::CaskDependent
```

since `cask_dependent` is never required.

**Comment by MikeMcQuaid** [comment]
@reitermarkus oops! See https://github.com/Homebrew/brew/pull/8138

**Pull request merged**